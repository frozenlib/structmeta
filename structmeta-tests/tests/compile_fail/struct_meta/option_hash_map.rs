use std::collections::HashMap;
use syn::LitInt;

#[derive(structmeta::StructMeta)]
struct Example {
    rest: Option<HashMap<String, LitInt>>,
}

fn main() {}
