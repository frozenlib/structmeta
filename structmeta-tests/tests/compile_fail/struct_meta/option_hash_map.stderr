error[E0277]: the trait bound `HashMap<String, LitInt>: Parse` is not satisfied
 --> tests/compile_fail/struct_meta/option_hash_map.rs:6:18
  |
6 |     rest: Option<HashMap<String, LitInt>>,
  |     ----         ^^^^^^^^^^^^^^^^^^^^^^^ the trait `Parse` is not implemented for `HashMap<String, LitInt>`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the following other types implement trait `Parse`:
            Example
            std::boxed::Box<T>
            proc_macro2::TokenStream
            proc_macro2::TokenTree
            proc_macro2::Group
            proc_macro2::Punct
            syn::Ident
            proc_macro2::Literal
          and $N others
note: required by a bound in `ParseBuffer::<'a>::parse`
 --> $CARGO/syn-2.0.29/src/parse.rs
  |
  |     pub fn parse<T: Parse>(&self) -> Result<T> {
  |                     ^^^^^ required by this bound in `ParseBuffer::<'a>::parse`
